server/tserver/src/main/java/org/apache/accumulo/tserver/metrics/TabletServerMetricsUtil.java

```diff
@@ -101,6 +101,14 @@ public int getMinorCompactionsQueued() {
     return result;
   }
 
+  public int getOnDemandOnlineCount() {
+    return tserver.getOnDemandOnlineCount();
```

Comment:
Other code in this class iterates over all of the online tablets to get the counts.  If this code did that then counter would not need to maintained (which can be tricky to keep correct, especially as the code changes).  Also it would make this code follow the pattern of the other metrics.  If this pattern of iterating over all the tablet is not good then it would probably be better to consider improving it for all metrics and not optimize a single metric in this PR.

Reply:
Updated in 6409f84 to use the set of online tablets. I removed the counter and the code that referenced it.